data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java

```diff
@@ -102,9 +102,14 @@ public PartitionRevokedHandler getPartitionRevokedHandler() {
     @Override
     public void recordReceived(final ConsumerRecord<?, ?> record) {
         final var tp = new TopicPartition(record.topic(), record.partition());
-        if (!offsetTrackers.containsKey(tp)) {
-            // Initialize offset tracker for the given record's topic/partition.
-            offsetTrackers.putIfAbsent(tp, new OffsetTracker(record.offset()));
+        final var offsetTracker = offsetTrackers.putIfAbsent(tp, new OffsetTracker(record.offset()));
+        if (offsetTracker != null && record.offset() < offsetTracker.initialOffset) {
+            if (record.offset() < offsetTracker.initialOffset) {
```

Comment:
`record.offset() < offsetTracker.initialOffset` is checked here twice

Reply:
:facepalm:

Reply:
Updated